### PR TITLE
Add Imaging Mass Cytometry Support to SpaceTx-format

### PR DESCRIPTION
**Pull request by ambrosejcarr**
- Adds a TileFormatter for Imaging Cytof
- Moves `postprocess_func` after the dummy codebook construction so that it can produce codebooks. 
- Adds a hook into starfish.data:

```python3
from starfish import data
exp = data.ImagingMassCytometry()
```

Testing plan: 
1.)
```bash
aws s3 sync  s3://spacetx.starfish.data.public/browse/raw/20181015/imaging_cytof/\
        BodenmillerBreastCancerSamples/ test_data
python3 examples/get_imc_data.py --input_dir test_data --output_dir test_data_formatted
```
2.) Load it up and take a look! Note that execution of cell 3 will break until #729 is merged, but that functionality is not critical to this testing. 

image

### CODE REVIEW

File: examples/get_imc_data.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+        self.file_path = file_path
+        self._tile_data = imread(self.file_path)
+
+    @property
+    def shape(self) -> Tuple[int, ...]:
+        return self._tile_data.shape
+
+    @property
+    def coordinates(self) -> Mapping[Union[str, Coordinates], Union[Number, Tuple[Number, Number]]]:
+        # TODO ambrosejcarr: ask about what these coordinates should correspond to.
```
Comment (ttung) [inline examples/get_imc_data.py]: going to defer to the wisdom of the person who made [this comment](https://github.com/spacetx/starfish/pull/647#discussion_r220971185) :)

File: examples/get_imc_data.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+            "HistoneH3(Yb176Di)",
+            "Ki-67(Er168Di)",
+            "PRAB(Gd158Di)",
+            "S6(Er170Di)",
+            "SMA(Nd148Di)",
+            "Twist(Nd145Di)",
+            "Vimentin(Dy162Di)",
+            "b-catenin(Ho165Di)",
+        ]
+        mapping = dict(zip(range(len(channels)), channels))
```
Comment (ttung) [inline examples/get_imc_data.py]: is this just `dict(zip(enumerate(channels)))`?

File: examples/get_imc_data.py [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+        - In Imaging Mass Cytometry, each channel specifies a unique target, so channel == target
+        - Imaging Mass Cytometry experiments have only one imaging round, round is hard coded as 1
+        - The spatial organization of the fields of view are not known to the starfish developers,
+          so they are filled by dummy coordinates
+
+        """
+        self.input_dir = input_dir
+
+    @property
+    def _ch_dict(self):
```
Comment (ttung) [inline examples/get_imc_data.py]:
```suggestion
    def _ch_dict(self) -> Mapping[int, str]:
```

File: examples/get_imc_data.py [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+            "S6(Er170Di)",
+            "SMA(Nd148Di)",
+            "Twist(Nd145Di)",
+            "Vimentin(Dy162Di)",
+            "b-catenin(Ho165Di)",
+        ]
+        mapping = dict(zip(range(len(channels)), channels))
+        return mapping
+
+    def ch_dict(self, item):
```
Comment (ttung) [inline examples/get_imc_data.py]:
```suggestion
    def ch_dict(self, ch: int) -> str:
```

File: examples/get_imc_data.py [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+        return self._ch_dict[item]
+
+    @property
+    def _fov_map(self):
+        fov_names = [
+            d for d in os.listdir(self.input_dir) if os.path.isdir(os.path.join(self.input_dir, d))
+        ]
+        mapping = dict(zip(range(len(fov_names)), fov_names))
```
Comment (ttung) [inline examples/get_imc_data.py]: enumerate

File: examples/get_imc_data.py [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+            "b-catenin(Ho165Di)",
+        ]
+        mapping = dict(zip(range(len(channels)), channels))
+        return mapping
+
+    def ch_dict(self, item):
+        return self._ch_dict[item]
+
+    @property
+    def _fov_map(self):
```
Comment (ttung) [inline examples/get_imc_data.py]: typing

File: examples/get_imc_data.py [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+        fov_names = [
+            d for d in os.listdir(self.input_dir) if os.path.isdir(os.path.join(self.input_dir, d))
+        ]
+        mapping = dict(zip(range(len(fov_names)), fov_names))
+        return mapping
+
+    def fov_map(self, item):
```
Comment (ttung) [inline examples/get_imc_data.py]:
```suggestion
    def fov_map(self, fov: int) -> str:
```

File: examples/get_imc_data.py (hunk 2 above)
Comment (ambrosejcarr) [reply examples/get_imc_data.py]: I have written the code you see here like 1000 times and never thought about using enumerate this way. It's actually even simpler!

`dict(enumerate(channels))`

File: examples/get_imc_data.py (hunk 4 above)
Comment (ambrosejcarr) [reply examples/get_imc_data.py]: Applied.

**Pull request merged**